From b42172ab954d9199f6f11deb60ddd0fc9d4be57c Mon Sep 17 00:00:00 2001
From: "Liu,Dalei" <daleix.liu@intel.com>
Date: Mon, 15 Jun 2015 11:21:13 +0800
Subject: [PATCH 424/462] ov8858: Delete V4L2_CID_TEST_PATTERN handling

This patch delete V4L2_CID_TEST_PATTERN handling to avoid
i2c error when system start.

Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-39349
Change-Id: I7cbd382e871629df19c56c187d53c800915c3066
Signed-off-by: Liu,Dalei <daleix.liu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/60526
Approver: Yunliang Ding <yunliang.ding@intel.com>
Reviewed-by: Yunliang Ding <yunliang.ding@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: IMIN (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/ov8858.c | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
index 9fcf171..4d0abc6 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
@@ -1770,7 +1770,6 @@ static int ov8858_s_ctrl(struct v4l2_ctrl *ctrl)
 	 * doesn't need to be taken here.
 	 */
 
-	/* We only handle V4L2_CID_RUN_MODE for now. */
 	switch (ctrl->id) {
 	case V4L2_CID_RUN_MODE:
 		switch (ctrl->val) {
@@ -1792,12 +1791,6 @@ static int ov8858_s_ctrl(struct v4l2_ctrl *ctrl)
 		dev->fps_index = 0;
 
 		return 0;
-	case V4L2_CID_TEST_PATTERN:
-		dev_dbg(&client->dev,
-			"%s: V4L2_CID_TEST_PATTERN = %d, val = 0x%04X\n",
-			__func__, V4L2_CID_TEST_PATTERN, ctrl->val);
-		return ov8858_write_reg(client, OV8858_16BIT,
-					OV8858_TEST_PATTERN_REG, ctrl->val);
 	case V4L2_CID_FOCUS_ABSOLUTE:
 		if (dev->vcm_driver && dev->vcm_driver->t_focus_abs)
 			return dev->vcm_driver->t_focus_abs(&dev->sd,
@@ -2099,13 +2092,6 @@ static const struct v4l2_ctrl_config ctrls[] = {
 		.flags = V4L2_CTRL_FLAG_READ_ONLY | V4L2_CTRL_FLAG_VOLATILE,
 	}, {
 		.ops = &ctrl_ops,
-		.id = V4L2_CID_TEST_PATTERN,
-		.name = "Test pattern",
-		.type = V4L2_CTRL_TYPE_INTEGER,
-		.step = 1,
-		.max = 0xffff,
-	}, {
-		.ops = &ctrl_ops,
 		.id = V4L2_CID_FOCUS_ABSOLUTE,
 		.name = "Focus absolute",
 		.type = V4L2_CTRL_TYPE_INTEGER,
-- 
1.9.1

